File: Data Structures & Algorithms/sliding-window-maximum/submission-0.go
func maxSlidingWindow(nums []int, k int) []int {
	dq := []int{}
	res := []int{}

	for i := 0; i < len(nums); i++ {
	
		for len(dq) > 0 && dq[0] <= i-k {
			dq = dq[1:]
		}

		
		for len(dq) > 0 && nums[dq[len(dq)-1]] < nums[i] {
			dq = dq[:len(dq)-1]
		}

		dq = append(dq, i)

		if i >= k-1 {
			res = append(res, nums[dq[0]])
		}
	}

	return res
}

File: Data Structures & Algorithms/subsets-ii/submission-0.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        nums.sort()

        res = []
        def dfs(i, curr):
            if i>=len(nums):
                res.append(curr.copy())
                return
            
            curr.append(nums[i])
            dfs(i+1, curr)
            curr.pop()
            x = 1
            for j in range(i+1, len(nums)):
                
                if nums[j]!=nums[i]:
                    break
                x+=1
            if x!=0:
                dfs(i+x, curr)
        
        dfs(0, [])
        return res
                


        